### PR TITLE
Wallabag plugin improvement

### PR DESCRIPTION
**Pull request by ArthurHoaro**
* Fixes a bug where URL weren't properly encoded.
  * Adds Wallabag V2 support.
  * Adds a URL function to handle trailing slash.
  * UT.
  * README updated.

### CODE REVIEW

File: plugins/wallabag/wallabag.php [hunk 1]
```diff
@@ -28,12 +30,20 @@ function hook_wallabag_render_linklist($data)
...
 
-    $wallabag_html = file_get_contents(PluginManager::$PLUGINS_PATH . '/wallabag/wallabag.html');
+    $version = isset($GLOBALS['plugins']['WALLABAG_VERSION']) ? $GLOBALS['plugins']['WALLABAG_VERSION'] : '';
+    $wallabagInstance = new WallabagInstance($GLOBALS['plugins']['WALLABAG_URL'], $version);
+
+    $wallabagHtml = file_get_contents(PluginManager::$PLUGINS_PATH . '/wallabag/wallabag.html');
 
     foreach ($data['links'] as &$value) {
-        $wallabag = sprintf($wallabag_html, $GLOBALS['plugins']['WALLABAG_URL'], $value['url'], PluginManager::$PLUGINS_PATH);
+        $wallabag = sprintf($wallabagHtml,
```
Comment (virtualtam) [inline plugins/wallabag/wallabag.php]: +newline:

```php
$wallabag = sprintf(
    [...]
);
```

File: tests/plugins/WallabagInstanceTest.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
```
Comment (virtualtam) [inline tests/plugins/WallabagInstanceTest.php]: class docstring should be just before class declaration

File: tests/plugins/WallabagInstanceTest.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+     */
+    function testWallabagInstanceV1() {
+        $instance = new WallabagInstance($this->instance, 1);
+        $expected = $this->instance . '/?plainurl=';
+        $result = $instance->getWallabagUrl();
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     *
```
Comment (virtualtam) [inline tests/plugins/WallabagInstanceTest.php]: empty docstrings

**Comment by ArthurHoaro** [comment]
I forgot about test docstrings, thanks for the review!

**Pull request merged**